Source/LuaBridge/RefCountedPtr.h

```diff
@@ -159,13 +159,17 @@ class RefCountedPtr : private detail::RefCountedPtrBase
     */
     void reset()
     {
-        if (m_p != 0)
+        auto myCount_it = getRefCounts().find(m_p); // use find() to avoid looking up m_p more than once
+        if (myCount_it != getRefCounts().end())
         {
-            if (--getRefCounts()[m_p] <= 0)
-                delete m_p;
-
-            m_p = 0;
+            if (--myCount_it->second <= 0)
+            {
+                if (m_p != 0)
```

Comment:
This check isn't required. You can safely delete a null pointer without any side effects.

Reply:
UPD. See another comment.

Reply:
Now the if is not longer required. Before it was required but the same check was missing at other locations, e.g. in the CTors.